### PR TITLE
[feature]: Clarified possibility to use redis-tls connection and added custom interview for this

### LINKED ISSUE #2562: [enhancement]: enable Redis TLS connection
### No existing issues.

- [x] There is no existing issue for my request.

### Description

_(shortened, was previously discussed on discord)_

Redis seems to support TLS (only) connection meanwhile, see https://github.com/redis/ioredis#tls-options

Also seems like ioBroker is basically able to do so also, but some details yet missing as `iob setup custom` fails

Docker log: 
`Error accepting a client connection: error:0A00010B:SSL routines::wrong version number (addr=172.24.0.1:52566 laddr=172.24.0.2:6379)`

ioBroker log: 
`The DB port 6379 is occupied by something that is not a Redis protocol server. Please check other software running on this port or, if you use iobroker, make sure to update to js-controller 2.0 or higher!`

Basic config example for Redis:

> port 0
> tls-port 6379
> tls-cert-file /var/ssl/server.crt
> tls-key-file /var/ssl/server.key
> tls-ca-cert-file /var/ssl/CA.crt
> tls-protocols "TLSv1.3"

### Why?

Encrypted communication also with objects/states database

### How?

https://github.com/redis/ioredis
https://ioredis.readthedocs.io/en/stable/README/#tls-options

### PR DESCRIPTION
**Pull request by foxriver76**
**Link the feature issue which is closed by this PR**

- closes #2562

**Implementation details**

The implementation allows to define the interview in the db class itself. It does not cover refactoring the existing questions.
For redis it is now asked if `tls` should be used, and if so, the certs are added to the config file. Downside: On change of cert file content you would need to rerun this.

**Tests**
- [ ] I have added tests to test this feature
- [x] It is not possible to test this feature

**Documentation**

- [ ] I have documented the new feature

Not sure if we should add docs for this @Apollon77? Or if the cli questions are enough. 

**If no tests added, please specify why it was not possible**

Really hard to implement as it requires user input, as long as no issues with it, lets skip tests here and add to avoid recursion on specific parts later

### CODE REVIEW

File: packages/common/src/lib/common/interview.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+ * The default redis interview shared between objects and states
+ *
+ * @param options config and type information
+ */
+export async function defaultRedisInterview<TConfig extends SharedDatabaseOptions>(
+    options: DefaultRedisInterviewOptions<TConfig>
+): Promise<TConfig> {
+    const { type, config } = options;
+
+    let answer = rl.question(`Dou you use a TLS connection for your "${type}" redis-server? [y/N]:`, {
```
Comment (Apollon77) [inline packages/common/src/lib/common/interview.ts:26]:
```suggestion
    let answer = rl.question(`Do you use a TLS connection for your "${type}" redis-server? [y/N]:`, {
```

File: packages/common/src/lib/common/interview.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        defaultInput: 'N'
+    });
+
+    if (answer.toLowerCase() === 'n') {
+        return config;
+    }
+
+    config.options.tls = {};
+
+    answer = rl.question(`Dou you use a self-signed certificate for your "${type}" redis-server? [y/N]:`, {
```
Comment (Apollon77) [inline packages/common/src/lib/common/interview.ts:37]:
```suggestion
    answer = rl.question(`Do you use a self-signed certificate for your "${type}" redis-server? [y/N]:`, {
```

File: packages/common/src/lib/common/interview.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        console.warn(`Could not read the "certificate" file, cert will be left empty: ${e.message}`);
+    }
+
+    answer = rl.question(`Please specify the path to your "${type}" redis-server "key" file:`);
+
+    try {
+        const keyContent = await fs.readFile(answer, { encoding: 'utf8' });
+        config.options.tls.key = keyContent;
+    } catch (e) {
+        console.warn(`Could not read the "key" file, cert will be left empty: ${e.message}`);
```
Comment (Apollon77) [inline packages/common/src/lib/common/interview.ts:61]: Ok makes it sense here to continue?

File: packages/common/src/lib/common/interview.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    }
+
+    answer = rl.question(`Please specify the path to your "${type}" redis-server "CA" file:`);
+
+    try {
+        const caContent = await fs.readFile(answer, { encoding: 'utf8' });
+        config.options.tls.ca = caContent;
+    } catch (e) {
+        console.warn(`Could not read the "CA" file, cert will be left empty: ${e.message}`);
```
Comment (Apollon77) [inline packages/common/src/lib/common/interview.ts:70]: same here

File: packages/common/src/lib/common/interview.ts [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        config.options.tls.rejectUnauthorized = false;
+    }
+
+    answer = rl.question(`Please specify the path to your "${type}" redis-server "certificate" file:`);
+
+    try {
+        const certContent = await fs.readFile(answer, { encoding: 'utf8' });
+        config.options.tls.cert = certContent;
+    } catch (e) {
+        console.warn(`Could not read the "certificate" file, cert will be left empty: ${e.message}`);
```
Comment (Apollon77) [inline packages/common/src/lib/common/interview.ts:52]: here too

**Review by Apollon77 — CHANGES_REQUESTED** [review]

File: packages/common/src/lib/common/interview.ts (hunk 4 above)
Comment (foxriver76) [reply packages/common/src/lib/common/interview.ts:70]: here it is already last question so doesn't matter

**Review by Apollon77 — APPROVED** [review]

**Pull request merged**